src/app/multi-editor/multi-editor.component.ts

```diff
@@ -137,9 +136,17 @@ export class MultiEditorComponent implements OnInit {
   }
 
   previewActions() {
+    if (!this.schemaKeysStoreService.validateActionsKeypaths(this.userActions)) {
```

Comment:
Could be better to just disable preview button instead maybe if it's invalid?

Also seems to me that you could make use of `@angular/forms` https://angular.io/guide/form-validation, better not to reinvent the wheel in a worse way, please check if forms could be used for your case, I think so though.